airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py

```diff
@@ -102,7 +103,29 @@ def records_with_state(records, state, stream_mapping, state_cursor_paths) -> It
                 # try second time as an absolute path in state file (i.e. bookmarks -> stream_name -> column -> value)
                 state_value = cursor_field.parse(record=state, path=state_cursor_paths[stream_name])
             except KeyError:
-                continue
+                # LEGACY states in incremental mode
+                if "cdc_state" in state.keys():
```

Comment:
I'm not sure this is the right place to address the CDC state.

My guiding principle here is: The Airbyte protocol and a result the CAT's should not be aware of how an individual connector consumes state.

As I understand `cdc_state` is specific to postgres's implementation (or database implementations). Which violates the above principle.

As an alternative, `source-postgres` should own this if statement.

I believe `source-sentry` [has implemented a similar method](https://github.com/airbytehq/airbyte/pull/21864/files#diff-350b5fdafcf9ae2b90ed4ab49c4d8d323826a10f16054941f5c812da040f7712R98) to handle legacy issues by overriding cursor value in certain situations.